app/js/views/master-view.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'components/react';
+
+import NavigationMenu from 'js/views/navigation-menu';
+
+const DEFAULT_HEADER_CONFIG = Object.freeze({
+  visible: true
+});
+
+const DEFAULT_FOOTER_CONFIG = Object.freeze({
+  visible: true
+});
+
+export default class MasterView extends React.Component {
```

Comment:
It's going to do more than just layout (manage modals for example), but I don't really care about naming - let's make it ```LayoutView``` then :)